### ISSUE TITLE
raw type error 

### ISSUE THREAD

**Issue by kellyngkq** [nus-cs2030/2122-s2#140]
Optional get(Object key) {
    return assessments.get(key);
}

Module.java:23: warning: [rawtypes] found raw type: Optional
    Optional get(Object key) {
    ^
  missing type arguments for generic class Optional<T>
  where T is a type-variable:
    T extends Object declared in class Optional
1 warning
Grading terminated.

Why do i get this error message and how can I fix it?

**Comment by michael192002** [comment]
I think you need to write a generic type, i.e. <T> for Optional

**Comment by kellyngkq** [comment]
do you mean doing Optional<T> get(Object key) ? 

Because when I do this, I still can't compile and will get this error:

Module.java:22: error: cannot find symbol
    Optional<T> get(Object key) {
             ^
  symbol:   class T
  location: class Module

**Comment by michael192002** [comment]
`Optional<T>` is needed, because if you only include Optional, then there is no type parameter specified

**Comment by kellyngkq** [comment]
not sure why the brackets are not showing, but that’s what I did and got the error message. i imported all the packages too

**Comment by michael192002** [comment]
Did you replace `Optional <T>` with the appropriate type parameter, e.g. `Optional<String> method(...)`?

**Comment by kellyngkq** [comment]
No, so that might be the problem. I thought putting T would mean putting a generic type, not sure why I have to specify?

**Comment by michael192002** [comment]
I think you only need to put a generic type when you are specifying in the generic class, but since Optional is already pre-built, I don't think you need to declare `<T>`. Similar case with ImList whereby we usually specify `ImList<Integer>` outside the ImList class instead of `ImList<T>`. Hope it helps!

**Comment by kellyngkq** [comment]
ohh I get it now
thank you !!
